Adding a property that should fail in the Wheel Brake System as a benchmark

vaibhavbsharma opened the pull request:
This version of Wheel Brake System has been part of the SPF repository
for the past few years. It can be found at https://github.com/SymbolicPathFinder/jpf-symbc/blob/master/src/examples/WBS.java. In this commit, I am dding it to the SV-COMP repository of benchmarks so that other tools can also benefit from it.

- WBS_prop1 is the name of the newly added benchmark
- sym-link to the jpf-regression license file has been added
- contributed by developers of Java Ranger (https://github.com/vaibhavbsharma/java-ranger). This benchmark originates from the WBS version maintained in the src/examples directory of the source code of SPF as mentioned earlier.

- The ReachSafety.set category has been updated to now include benchmarks in the java-ranger-regression directory
- assert.prp should be used to verify the intended property
- expected answer has been added WBS_prop1.yml

mmuesly commented on java/java-ranger-regression/WBS_prop1/Main.java:
```diff
@@ -0,0 +1,305 @@
+
+	public static void main(String[] args) {
+		Main Main = new Main();
+		Main.update(Verifier.nondetInt(), Verifier.nondetBoolean(), Verifier.nondetBoolean());
```
Is there a specific reason, why this function is called 10 times and this is not done in a for loop?
Might you add a comment?

Otherwise, this seems to be code supposed to run in a while(true)-loop. If that is the case, I would advocate for a second version of this test case with a while-loop.

vaibhavbsharma replied:
There's no specific reason for not using a loop. I've fixed this with a loop in this test case itself.

mmuesly commented on java/java-ranger-regression/WBS_prop1/Main.java:
```diff
@@ -0,0 +1,305 @@
+				   }
+			   }
+		   }
+		//Debug.printPC("PC @fter(PadalPos == 0) [Region 1] = ");
```
Nit-Pick:
I think, it might be nice to remove this code as the Debug class is not part of the benchmark.

vaibhavbsharma replied:
Done. I've also removed all the other commented out lines in this test case.

mmuesly commented on java/java-ranger-regression/WBS_prop1/Main.java:
```diff
@@ -0,0 +1,305 @@
+
+	   WBS_Node_WBS_BSCU_Command_Is_Normal_Relational_Operator = (WBS_Node_WBS_BSCU_SystemModeSelCmd_Unit_Delay == 0); 
+
+	   if ((PedalPos == 0)) {
```
I personally think, it might be interesting to distribute the very same benchmark file with a switch-case statement instead of this if-then-else cascade in order to integrate some more features of the Java language. From my point of view, this seems to be the ideal candidate to use this test case in two descriptions.

vaibhavbsharma replied:
I agree. I will add this test case too into the repo soon. I've currently also got quite a few other benchmarks to add and will add this into my list.

mmuesly commented on java/java-ranger-regression/WBS_prop1/Main.java:
```diff
@@ -0,0 +1,305 @@
+	   WBS_Node_WBS_BSCU_SystemModeSelCmd_rlt_PRE = Sys_Mode;
+//	   WBS_Node_WBS_BSCU_SystemModeSelCmd_rlt_PRE += WBS_Node_WBS_BSCU_SystemModeSelCmd_Logical_Operator6 ? 1 : 0;
+
+		// Assertions added by MWW: these are the truth values for "my" version of the model - may not be true for this code, but should be
```
It seems like all these assert should explicitly not be checked.
Might it be worth to remove them in this case to keep the test case clean?

vaibhavbsharma replied:
Done.

vaibhavbsharma commented:
I've made changes to this benchmark based on discussions in #860. Please let me know if anything else is required for this benchmark to be merged.

mmuesly commented on java/java-ranger-regression/WBS/WBS_prop1.yml:
```diff
@@ -0,0 +1,8 @@
+format_version: "1.0"
+input_files:
+  - ../../common/
+  - impl/
```
It feels like the `java/java-ranger-regression/WBS/prop1/Main.java`
should be included here as well, if I am not totally mistaken.

vaibhavbsharma replied:
Done. I was misunderstanding this `input_files` directive but I've now added `prop1/` as an `input_files` entry.

mmuesly commented on java/java-ranger-regression/WBS/prop1/Main.java:
```diff
@@ -0,0 +1,18 @@
+		  AutoBrake = Verifier.nondetBoolean();
+		  Skid = Verifier.nondetBoolean();
+		  wbs.update(PedalPos, AutoBrake, Skid);
+		  // This assertion should fail:
```
This comment is contradicting the expected verdict. Please remove the comment line or change the label accordingly.

vaibhavbsharma replied:
Fixed. Thanks for catching this.

vaibhavbsharma commented:
Thanks for looking at my PR, @mmuesly!! I've addressed your comments and also integrated the 3 additional properties from #860, #861, and #862 into this pull request.

mmuesly commented:
@vaibhavbsharma I think you are still missing to remove this file: https://github.com/sosy-lab/sv-benchmarks/blob/master/java/java-ranger-regression/WBS_prop4/Main.java

Once that is done, this PR should be ready to merge.
